src/Bundle/Migration/CodeGenerator.php

```diff
@@ -0,0 +1,183 @@
+    {
+        $classConfiguration = ['stmts' => $body];
+
+        if ($extends !== null) {
+            $classConfiguration['extends'] = new Identifier($extends);
+        }
+
+        if (count($implements) > 0) {
+            $classConfiguration['implements'] = array_map(
+                static fn (string $interfaceName) => new Identifier($interfaceName),
+                $implements,
+            );
+        }
+
+        $this->classNodes[] = new Class_(new Identifier($className), $classConfiguration);
+    }
+
+    public function generateGetResourceClassFunction(string $resourceNameOrString): Node
+    {
+        // If it's a class generate something like: Order::class
+        if (class_exists($resourceNameOrString)) {
+            $returnStatement = new ClassConstFetch(
+                $this->getRelativeClassName($resourceNameOrString),
+                'class',
+            );
+        } else {
+            // Otherwise just return the string as is
+            $returnStatement = new String_($resourceNameOrString);
+        }
+
+        return new ClassMethod(
+            new Identifier('getResourceClass'),
+            [
+                'flags' => Class_::MODIFIER_PUBLIC,
+                'returnType' => 'string',
+                'stmts' => [
+                    new Return_($returnStatement),
+                ],
+            ],
+        );
+    }
+
+    public static function createStaticFunction(string $functionName, string $returnValue): Node
+    {
+        return new ClassMethod(
+            new Identifier($functionName),
+            [
+                'flags' => Class_::MODIFIER_PUBLIC | Class_::MODIFIER_STATIC,
+                'returnType' => 'string',
+                'stmts' => [
+                    new Return_(new String_($returnValue)),
+                ],
+            ],
+        );
+    }
+
+    public static function createFunction(
+        string $functionName,
+        array $params,
+        array $body,
+    ): Node {
+        return new ClassMethod(
+            new Identifier($functionName),
+            [
+                'flags' => Class_::MODIFIER_PUBLIC,
+                'returnType' => 'void',
+                'params' => array_map(
+                    fn (array $param) => new Param(new Variable($param[1]), null, $param[0]),
+                    $params,
+                ),
+                'stmts' => $body,
+            ],
+        );
+    }
+
+    public function build(): string
+    {
+        $nodes = [];
+        if ($this->namespace !== null) {
+            $nodes[] = $this->namespace;
+        }
+
+        foreach ($this->getUseStatementNodes() as $useStatements) {
+            $nodes[] = $useStatements;
+        }
+
+        foreach ($this->classNodes as $classNode) {
+            $nodes[] = $classNode;
+        }
+
+        $generatedCode = "<?php\n" . $this->codeOutputter->prettyPrint($nodes);
+
+        $this->classNodes = [];
```

Comment:
is it to reset values?

Reply:
Yes the code generator is stateful sadly. The reason for this is that we need a way to accumulate the list of import statements. In order to ensure that when the generator is used on multiple grids, we need to reset the code generator.